### PR TITLE
Fix UID management for ubuntu users

### PR DESCRIPTION
**Pull request by rbren**
**Short description of the problem this fixes or functionality that this introduces. This may be used for the CHANGELOG**

Attempting to fix https://github.com/All-Hands-AI/OpenHands/issues/3921

This will need some serious testing, esp on Ubuntu. @tobitege should be able to help

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

---
**Link of any specific issues this addresses**

### CODE REVIEW

File: openhands/runtime/client/client.py [hunk 1]
```diff
@@ -115,107 +115,34 @@ async def ainit(self):
...
-            - If the user doesn't exist, it proceeds to create the user.
-        * Sudo Configuration:
-            - Appends %sudo ALL=(ALL) NOPASSWD:ALL to /etc/sudoers to grant
-              passwordless sudo access to the sudo group.
-            - Adds the user to the sudo group with the useradd command, handling
-              UID conflicts by incrementing the UID if necessary.
-        """
-
-        # First create the working directory, independent of the user
-        logger.info(f'Client working directory: {self.initial_pwd}')
```
Comment (tobitege) [inline openhands/runtime/client/client.py:137]: this whole section is important to be kept!

File: openhands/runtime/client/client.py [hunk 2]
```diff
@@ -115,107 +115,34 @@ async def ainit(self):
...
-            if output.returncode == 0:
-                logger.debug(
-                    f'Added user `{username}` successfully with UID {user_id}. Output: [{output.stdout.decode()}]'
-                )
-                break
-            elif f'UID {user_id} is not unique' in output.stderr.decode():
-                logger.warning(
-                    f'UID {user_id} is not unique. Incrementing UID and retrying...'
-                )
-                user_id += 1
```
Comment (tobitege) [inline openhands/runtime/client/client.py:214]: This line and the "while" loop line are likely the things that need to be removed only, imo.
If user with _same name and id_ already exists, that's fine.
Only if the existing username is different than "openhands", we should abort.

Comment (rbren) [reply openhands/runtime/client/client.py:214]: OK hopefully this is better

**Comment by tobitege** [comment]
This _should_ work fine now, thanks!

**Review by tobitege — APPROVED** [review]
LGTM, thanks for the fast fix!

**Pull request merged**